### PR TITLE
Adjust stock movement label when product is virtual

### LINKED ISSUE #30331: The label "Shipped products" is not correct in the movement section when ordering a virtual product
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Previously reported here: https://github.com/PrestaShop/PrestaShop/pull/28640#pullrequestreview-1177295537
The label "Shipped products" is not correct in the movement section when ordering a virtual product

### Expected behavior

Change the wording to "Sold product"

### Steps to reproduce

1. Go to BO > Advanced Parameters > New & Experimental Features page & Enable New product page - Single store
2. Create a vector product with downloadable file & quantity = 10 for example
3. Create an order with this product (No shipping is selected)
4. Go to BO > Orders > View the last order created & change the status to "Payment accepted"

The customer, in this case, can download his attachment > I think, in this specific case, we should add a line for the stock movement

5. Change the Order status to shipped 
6. Go to BO > Catalog > Product list V2 > Edit the product created
7. Check the Virtual tab > The expression "Shipped" is not correct with a virtual product

image

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

8.0

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | https://github.com/PrestaShop/PrestaShop/issues/30331
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | https://github.com/PrestaShop/PrestaShop/issues/30331
| Fixed ticket?     | Fixes #30331
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

File: src/Core/Domain/Product/Stock/Query/GetProductStockMovements.php [hunk 1]
```diff
@@ -64,11 +61,6 @@ class GetProductStockMovements
      */
     private $productId;
 
-    /**
-     * @throws ProductConstraintException
-     * @throws ShopException
-     * @throws LogicException
-     */
```
Comment (zuk3975) [inline src/Core/Domain/Product/Stock/Query/GetProductStockMovements.php:71]: unrelated change, just noticed these docs are outdated, constructor doesn't throw any of these exceptions and all arguments are typed so I removed the docblock completely

File: src/PrestaShopBundle/Form/Admin/Sell/Product/Stock/StockType.php [hunk 2]
```diff
@@ -98,13 +98,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'class' => 'product_packed_products',
                     'data-reference-label' => $this->trans('Ref: %s', 'Admin.Catalog.Feature'),
                 ],
-                'min_length' => 3,
+                'min_length' => 4,
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Sell/Product/Stock/StockType.php]: Why change this one? I think 3 was set on purpose

Comment (zuk3975) [reply src/PrestaShopBundle/Form/Admin/Sell/Product/Stock/StockType.php]: just testing, forgot to remove the change :+1:

**Review by jolelievre — APPROVED** [review]
Thanks @zuk3975

**Review by kpodemski — APPROVED** [review]

**Review by MhiriFaten — APPROVED** [review]
Hello @zuk3975 ,

I have checked your PR, the issue is well fixed and the wording is corrected as expected !
The automated tests are OK.

It's QA approved :heavy_check_mark: 

Thank you !

**Pull request merged**